network/framework/src/application/tests.rs

```diff
@@ -427,6 +428,76 @@ fn test_peers_and_metadata_caching() {
     );
 }
 
+#[tokio::test]
+async fn test_peers_and_metadata_subscriptions() {
+    // Create the peers and metadata container
+    let network_ids = vec![NetworkId::Validator, NetworkId::Vfn];
+    let peers_and_metadata = PeersAndMetadata::new(&network_ids);
+
+    let mut connection_events = peers_and_metadata.subscribe();
+
+    match connection_events.try_recv() {
+        Ok(wat) => {
```

Comment:
nit: `wat` -> `event`? Or, maybe I'm missing what `wat` is short for 😄

Reply:
`https://www.destroyallsoftware.com/talks/wat`
but I'll put a better variable name here